src/credentials/DeviceAttestationVendorReserved.h

```diff
@@ -0,0 +1,218 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/core/CHIPTLV.h>
+#include <lib/support/Span.h>
+
+#include <cstdint>
+
+namespace chip {
+namespace Credentials {
+
+struct VendorReservedElement
+{
+    uint16_t vendorId;
+    uint16_t profileNum;
+    uint32_t tagNum;
+    ByteSpan vendorReservedData;
+};
+
+// extract elements out of the device attestation bytespan
+class DeviceAttestationVendorReservedDeconstructor
+{
+
+public:
+    DeviceAttestationVendorReservedDeconstructor() {}
+
+    // read TLV until first profile tag
+    CHIP_ERROR PrepareToReadVendorReservedElements(const ByteSpan & attestationElements, size_t count)
+    {
+        mNumVendorReservedData = count;
```

Comment:
Set `mIsInitialized` to false at start of function to support restarts that are not in inconsistent state if the early-check error handling hits on a second run.